src/spring-cloud/azext_spring_cloud/gateway.py

```diff
@@ -0,0 +1,232 @@
+
+def gateway_clear(cmd, client, resource_group, service):
+    gateway = client.gateways.get(resource_group, service, DEFAULT_NAME)
+    properties = models.GatewayProperties()
+    sku = models.Sku(name=gateway.sku.name, tier=gateway.sku.tier)
+    gateway_resource = models.GatewayResource(properties=properties, sku=sku)
+    return client.gateways.begin_create_or_update(resource_group, service, DEFAULT_NAME, gateway_resource)
+
+
+def gateway_custom_domain_show(cmd, client, resource_group, service, domain_name):
+    return client.gateway_custom_domains.get(resource_group, service, DEFAULT_NAME, domain_name)
+
+
+def gateway_custom_domain_list(cmd, client, resource_group, service):
+    return client.gateway_custom_domains.list(resource_group, service, DEFAULT_NAME)
+
+
+def gateway_custom_domain_update(cmd, client, resource_group, service,
+                                 domain_name,
+                                 certificate=None):
+    properties = models.GatewayCustomDomainProperties()
+    if certificate is not None:
+        certificate_response = client.gateways.certificates.get(
+            resource_group, service, certificate)
+        properties = models.GatewayCustomDomainProperties(
+            thumbprint=certificate_response.properties.thumbprint
+        )
+
+    custom_domain_resource = models.GatewayCustomDomainResource(
+        properties=properties)
+    return client.gateway_custom_domains.begin_create_or_update(resource_group, service, DEFAULT_NAME,
+                                                                domain_name, custom_domain_resource)
+
+
+def gateway_custom_domain_unbind(cmd, client, resource_group, service, domain_name):
+    client.gateway_custom_domains.get(resource_group, service,
+                                      DEFAULT_NAME, domain_name)
+    return client.gateway_custom_domains.begin_delete(resource_group, service, DEFAULT_NAME, domain_name)
+
+
+def gateway_route_config_show(cmd, client, resource_group, service, name):
+    return client.gateway_route_configs.get(resource_group, service, DEFAULT_NAME, name)
+
+
+def gateway_route_config_list(cmd, client, resource_group, service):
+    return client.gateway_route_configs.list(resource_group, service, DEFAULT_NAME)
+
+
+def gateway_route_config_create(cmd, client, resource_group, service, name,
+                                app_name=None,
+                                routes_json=None,
+                                routes_file=None):
+    if routes_json is not None and routes_file is not None:
+        raise CLIError(
+            "You can only specify either --routes-json or --routes-file.")
+
+    route_configs = client.gateway_route_configs.list(
+        resource_group, service, DEFAULT_NAME)
+    if name in (route_config.name for route_config in list(route_configs)):
+        raise CLIError("Route config " + name + " already exists")
+
+    app_resource = client.apps.get(resource_group, service, app_name)
+
+    routes = []
```

Comment:
What if `routes_json` and `routes_file` both provided? I would suggest we reject that case for the first version. And we can save effort to explain what will happen when the two params are given